compile/plugin.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python3
+
+import os
```

Comment:
It might make more sense to have this file (and template) under `scripts` since it is only used when we generate from proto.
Unless that stops tests from reading this file.